src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -5,17 +5,20 @@
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.Handler;
-import android.os.Message;
+
 import android.util.Log;
 
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
 import org.thaliproject.p2p.btconnectorlib.BTConnector;
 import org.thaliproject.p2p.btconnectorlib.BTConnectorSettings;
 import org.thaliproject.p2p.btconnectorlib.ServiceItem;
 
-import java.net.ServerSocket;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+
 
 /**
  * Created by juksilve on 14.5.2015.
```

Comment:
On line 28 below you set context to null. Which is fine but since context is only set in the constructor you can make it final. It's always good habit in multi-threaded code to make anything that can be final, into final, it prevents mistakes. So i would review all your state variables and see which ones you can make final.